govuk-paas/paas-rds-broker
# [#118479861] Seed and rotate RDS master password

# What

Story: [Generating secure master password for Postgres Instances](https://www.pivotaltracker.com/story/show/118479861)

We want to add a seed to generate the DB instances master passwords so they can't easily be guessed. The seed is set using a broker property.
As a consequence, we need to be able to rotate the passwords when the seed is changed. This is implemented by checking the connection on all instances at startup and resetting the password if necessary.
This relies on tags being applied on broker created instances. A temporary task was added to the pipeline to tag existing instances.

This should be tested with PRs on paas-aws-broker-boshrelease, paas-cf and paas-docker-cloudfoundry-tools.
# How to review
- Deploy on an existing environment. It should detect the RDS passwords have changed and reset the passwords.
- Check the rds broker logs to confirm
- Bind an existing DB
- Create and bind a new DB
- Unbind
- Run autodelete
### Expected error when the password is wrong

This should no happen:

```
$ cf bind-service mtest healthcheck-db
Binding service healthcheck-db to app mtest in org testers / space healthcheck as admin...
FAILED
Server error, status code: 502, error code: 10001, message: Service broker error: pq: password authentication failed for user "DJ4YDF83aJXWxX26"
```
# Who can review

Anyone but @saliceti, @mtekel, @keymon
